{
  "test_block_invalid_put": {
        "title": "------------------[Block Placement Test: Invalid-Sized File]------------------",
        "description": "This test validates that a tapestry.Block object will not accept a put() request for a file that is larger than the remaining size of the block.",
        "pass message": "[PASS] The Block object indicated it rejected the file.",
        "fail message": "[FAIL] See Error:"
    },
  "test_verify_invalid_block": {
        "title": "------------------[Block Signature Validation Test]------------------",
        "description": "This test validates that tapestry will reject a file with invalid signatures",
        "pass message": "[PASS] The verify_blocks correctly ignored the invalidly-signed file.",
        "fail message": "[FAIL] See Error:"
    },
  "test_recovery_index_invalid": {
        "title": "------------------[Recovery Index Confusion Test]------------------",
        "description": "Makes sure that RecoveryIndexError is raised correctly when loading a tapestry.RecoveryIndex option with data.",
        "pass message": "[PASS] Intercepted the expected exception.",
        "fail message": "[FAIL] See Error:"
    },
  "test_decompress_uncompressed": {
        "title": "------------------[Task_Decompress Test: Uncompressed Target]------------------",
        "description": "This test validates that the expected response is returned by TaskDecompress when it is called against a target file that does not contain BZ2 compressed data.",
        "pass message": "[PASS] The file was correctly skipped.",
        "fail message": "[FAIL] See Error:"
    },
    "test_TaskCheckIntegrity": {
        "title": "----------------------------[Integrity Check Test]-----------------------------",
        "description": "Feeds a garbage hash to TaskCheckIntegrity to ensure that files are correctly rejected by this test when necessary.",
        "pass message": "[PASS] The hashing mismatch was correctly detected",
        "fail message": "[FAIL] One or more errors were raised in testing:"
    },
    "test_sftp_connect_invalid": {
        "title": "-------------------------[SFTP Connection Trust Test]---------------------------",
        "description": "Calls sftp_connect against a known-good SFTP server using an invalid trust, and ensures a valid response came back.",
        "pass message": "[PASS] sftp_connect refused to connect to an untrusted server.",
        "fail message": "[FAIL] One or more errors were raised in testing:"
    },
    "test_sftp_connect_down": {
        "title": "-----------------------[SFTP Connection Failure Test]--------------------------",
        "description": "Calls sftp_connect against a non-extant SFTP server and ensures the expected responses come back",
        "pass message": "[PASS] sftp_connect behaves correctly when the target server is down.",
        "fail message": "[FAIL] One or more errors were raised in testing:"
    },
    "test_sftp_place": {
        "title": "----------------------------[SFTP Placement Test]------------------------------",
        "description": "Attempts to place a file onto an SFTP server without permissions and observes results.",
        "pass message": "[PASS] sftp_place behaves as expected when write permissions are absent.",
        "fail message": "[FAIL] One or more errors were raised in testing:"
    },
    "test_sftp_find": {
        "title": "---------------------------[SFTP File Listing Test]----------------------------",
        "description": "Calls sftp_find to obtain a listing of all files in the target directory on the SFTP without adequate permissions.",
        "pass message": "[PASS] sftp_find behaves as expected when write permissions are missing.",
        "fail message": "[FAIL] One or more errors were raised in testing:"
    },
    "test_sftp_fetch": {
        "title": "--------------------------[SFTP File Retrieval Test]---------------------------",
        "description": "Attempts to fetch a known-to-exist file from the SFTP share without appropriate permission.",
        "pass message": "[PASS] sftp_fetch behaves as expected on encountering an error.",
        "fail message": "[FAIL] One or more errors were raised in testing:"
    }
}